### DISCUSSION TITLE
proxy_ssl_server_name error 【upstream SSL certificate does not match "xxxx"】

### DISCUSSION THREAD

**Discussion by zerotocode398** [nginx/nginx#328] [Q&A] [answered]
# nginx version
![Image](https://github.com/user-attachments/assets/bcbfd69b-3722-455d-9d68-d1eb98c8f7df)
# nginx config
![Image](https://github.com/user-attachments/assets/5542c0c9-4609-4f6b-9e0c-ecd7e079bf6c)
# error log
![Image](https://github.com/user-attachments/assets/536ee797-6e99-40cf-9b92-eed73544c53a)

Why did this error occur and how should it be resolved? 
Can anyone help me? Thank you

**Comment by viktorberezikov** [comment, marked as answer]
You have configured nginx to verify the upstream certificate with `proxy_ssl_verify on;`. At the same time, you expect the upstream certificate to be issued for the `proxy_ssl_name partner_self_service_platform;` domain name. The error indicates that the certificate is issued for a different domain name.
You can investigate which certificate your upstream contains with the openssl utility.
For example:
`echo Q | openssl s_client -connect www.google.com:443 -servername www.google.com`
